.github/workflows/publish.yml

```diff
@@ -15,7 +15,7 @@ jobs:
       fail-fast: false
       matrix:
         job:
-          - { target: x86_64-unknown-linux-gnu, file_suffix: amd64-linux, os: ubuntu-latest}
+          - { target: x86_64-unknown-linux-gnu, file_suffix: amd64-linux, os: ubuntu-22.04}
```

Comment:
- Do you want to pin the macos and windows builds too?
- Are we *actually* building x86_64-apple-darwin? Or is it really arm

Reply:
maybe? what do you propose pinning them to?

And yes it is actually arm-apple-darwin

Reply:
I think, per https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories, that they should be `windows-2022` and `macos-14` respectively. And we should probably fix up the target name for `x86_64-apple-darwin` since it is inaccurate. I think it is `aarch64-apple-darwin` but please check me on that.

Reply:
no wait, that target is correct, it's the Cargo triplet

Reply:
Yes, almost all recent macs are arm based, and the macos-14 builder is too, and `macos-latest` is `macos-14` so these were already arm builds.

Reply:
Yes, and Rosetta was helpfully running the x86 binary for me.